internal/server/middleware/grpc/middleware.go

```diff
@@ -88,33 +90,127 @@ type ResponseDurationRecordable interface {
 
 // EvaluationUnaryInterceptor sets required request/response fields.
 // Note: this should be added before any caching interceptor to ensure the request id/response fields are unique.
-func EvaluationUnaryInterceptor(ctx context.Context, req interface{}, _ *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (resp interface{}, err error) {
-	startTime := time.Now().UTC()
+func EvaluationUnaryInterceptor(analyticsEnabled bool) grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, _ *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (resp interface{}, err error) {
+		startTime := time.Now().UTC()
 
-	// set request ID if not present
-	requestID := uuid.Must(uuid.NewV4()).String()
-	if r, ok := req.(RequestIdentifiable); ok {
-		requestID = r.SetRequestIDIfNotBlank(requestID)
+		// set request ID if not present
+		requestID := uuid.Must(uuid.NewV4()).String()
+		if r, ok := req.(RequestIdentifiable); ok {
+			requestID = r.SetRequestIDIfNotBlank(requestID)
 
-		resp, err = handler(ctx, req)
-		if err != nil {
-			return resp, err
-		}
+			resp, err = handler(ctx, req)
+			if err != nil {
+				return resp, err
+			}
 
-		// set request ID on response
-		if r, ok := resp.(RequestIdentifiable); ok {
-			_ = r.SetRequestIDIfNotBlank(requestID)
-		}
+			// set request ID on response
+			if r, ok := resp.(RequestIdentifiable); ok {
+				_ = r.SetRequestIDIfNotBlank(requestID)
+			}
 
-		// record start, end, duration on response types
-		if r, ok := resp.(ResponseDurationRecordable); ok {
-			r.SetTimestamps(startTime, time.Now().UTC())
+			// record start, end, duration on response types
+			if r, ok := resp.(ResponseDurationRecordable); ok {
+				r.SetTimestamps(startTime, time.Now().UTC())
+			}
+
+			if analyticsEnabled {
+				span := trace.SpanFromContext(ctx)
+
+				switch r := resp.(type) {
+				case *evaluation.VariantEvaluationResponse:
+					// This "should" always be an evalution request under these circumstances.
+					if evaluationRequest, ok := req.(*evaluation.EvaluationRequest); ok {
+						evaluationResponses := []*analytics.EvaluationResponse{
+							{
+								NamespaceKey: evaluationRequest.GetNamespaceKey(),
+								FlagKey:      r.GetFlagKey(),
+								FlagType:     evaluation.EvaluationFlagType_VARIANT_FLAG_TYPE.String(),
+								Match:        r.GetMatch(),
+								Reason:       r.GetReason().String(),
+								Timestamp:    r.GetTimestamp().AsTime(),
+							},
+						}
+
+						if evaluationResponsesBytes, err := json.Marshal(evaluationResponses); err == nil {
+							keyValue := []attribute.KeyValue{
+								{
+									Key:   "flipt.evaluation.response",
+									Value: attribute.StringValue(string(evaluationResponsesBytes)),
```

Comment:
Here is a bit of a mad hack: https://pkg.go.dev/go.opentelemetry.io/otel/attribute#Stringer

You could add a `String() string` method to `*analytics.EvaluationResponse` that marshals out a Go string. Then instead of marshalling this to JSON and then unmarshalling again in the exporter, you could pass it here with `attributes.Stringer` and just cast it back to a `*analytics.EvaluationResponse` in the exporter.
Otel will take care of marshalling lazilly when it needs to go over the wire.

Reply:
@GeorgeMac Ahh I started to implement it this way just now and realized,

What is actually happening here is I am marshalling a `[]*EvaluationResponse` to JSON instead of just a single `*EvaluationResponse`.

I chose to do it this way because it helps with the `Batch` implementation.

Reply:
Could you re-type that slice for the same ends?
```go
type EvaluationResponses []*EvaluationResponse

func (r EvaliationResponses) String() string {
    v, _ := json.Marshal(r)
    return string(v)
}
```

And then use `analytics.EvaluationResponses{}` instead of `[]*analytics.EvaluationResponse`?
Either way, not important, just seeing if we can avoid marshalling back and forth.